src/realm/parser/driver.cpp

```diff
@@ -537,7 +537,7 @@ Query EqualityNode::visit(ParserDriver* drv)
     else if (right->has_single_value() && (left_type == right_type || left_type == type_Mixed)) {
         Mixed val = right->get_mixed();
         const ObjPropertyBase* prop = dynamic_cast<const ObjPropertyBase*>(left.get());
-        if (prop && !prop->links_exist()) {
+        if (prop && !prop->links_exist() && !prop->has_path()) {
```

Comment:
This is only managing simple values, right? No links?

Reply:
No links and no path.